CMakeLists.txt

```diff
@@ -1588,7 +1588,19 @@ if(HEX_FILES_TO_MERGE)
 endif()
 
 if(EMU_PLATFORM)
-  include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
+  if(NOT EMU_PLATFORM_TARGET)
+    if(NOT EMU_PLATFORM_DEFAULT)
+      list(GET EMU_PLATFORM 0 EMU_PLATFORM_DEFAULT)
+      message(WARNING "EMU_PLATFORM_DEFAULT is not set! Defaulting to "${EMU_PLATFORM_DEFAULT})
+    endif()
+    set(EMU_PLATFORM_TARGET ${EMU_PLATFORM_DEFAULT})
+  endif()
+
+  if(EXISTS "${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM_TARGET}.cmake")
```

Comment:
To me, this looks like the `EMU_PLATFORM_TARGET` will always be used respectively of whether the `EMU_PLATFORM` list contains that platform or not.

For example:
```
# Some board.cmake
set(EMU_PLATFORM qemu renode)
```
User executes:
```
cmake ... -DEMU_PLATFORM_TARGET=tsim
 ```
then this code
```
 include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM_TARGET}.cmake)
```
will execute as:
```
 include(${ZEPHYR_BASE}/cmake/emu/tsim.cmake)
```
because tsim.cmake exists, is that intentional ?

Reply:
Thanks for spotting that. I've added an additional check to test whether the `EMU_PLATFORM` list contains the value of `EMU_PLATFORM_TARGET`.